internal/services/apimanagement/api_management_policy_fragment_resource.go

```diff
@@ -0,0 +1,232 @@
+	subscriptionId := meta.(*clients.Client).Account.SubscriptionId
+	defer cancel()
+
+	id := policyfragment.NewPolicyFragmentID(subscriptionId, d.Get("resource_group_name").(string), d.Get("api_management_name").(string), d.Get("name").(string))
+	format := policyfragment.PolicyFragmentContentFormat(d.Get("format").(string))
+	if d.IsNewResource() {
+		existing, err := client.Get(ctx, id, policyfragment.GetOperationOptions{
+			Format: &format,
+		})
+		if err != nil {
+			if !response.WasNotFound(existing.HttpResponse) {
+				return fmt.Errorf("checking for presence of existing %s: %s", id, err)
+			}
+		}
+
+		if !response.WasNotFound(existing.HttpResponse) {
+			return tf.ImportAsExistsError("azurerm_api_management_policy_fragment", id.ID())
+		}
+	}
+
+	description := d.Get("description").(string)
+	value := d.Get("value").(string)
+
+	parameters := policyfragment.PolicyFragmentContract{
+		Properties: &policyfragment.PolicyFragmentContractProperties{
+			Description: pointer.To(description),
+			Format:      pointer.To(format),
+			Value:       value,
+		},
+	}
+
+	if err := client.CreateOrUpdateThenPoll(ctx, id, parameters, policyfragment.CreateOrUpdateOperationOptions{}); err != nil {
+		return fmt.Errorf("creating/updating %s: %+v", id, err)
+	}
+
+	d.SetId(id.ID())
+
+	return resourceApiManagementPolicyFragmentRead(d, meta)
+}
+
+func resourceApiManagementPolicyFragmentUpdate(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).ApiManagement.PolicyFragmentClient
+	ctx, cancel := timeouts.ForCreateUpdate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := policyfragment.ParsePolicyFragmentIDInsensitively(d.Get("id").(string))
+	if err != nil {
+		return err
+	}
+
+	format := policyfragment.PolicyFragmentContentFormat(d.Get("format").(string))
+
+	description := d.Get("description").(string)
+	value := d.Get("value").(string)
+
+	parameters := policyfragment.PolicyFragmentContract{
+		Properties: &policyfragment.PolicyFragmentContractProperties{
+			Description: pointer.To(description),
+			Format:      pointer.To(format),
+			Value:       value,
+		},
+	}
+
+	if err := client.CreateOrUpdateThenPoll(ctx, pointer.From(id), parameters, policyfragment.CreateOrUpdateOperationOptions{}); err != nil {
+		return fmt.Errorf("creating/updating %s: %+v", id, err)
+	}
+
+	d.SetId(id.ID())
+
+	return resourceApiManagementPolicyFragmentRead(d, meta)
+}
+
+func resourceApiManagementPolicyFragmentRead(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).ApiManagement.PolicyFragmentClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := policyfragment.ParsePolicyFragmentIDInsensitively(d.Id())
+	if err != nil {
+		return err
+	}
+
+	format := policyfragment.PolicyFragmentContentFormat(d.Get("format").(string))
+	resp, err := client.Get(ctx, *id, policyfragment.GetOperationOptions{
+		Format: &format,
+	})
```

Comment:
(as below) can we update this to use `DefaultGetOperationOptions` to handle any default values for options:

```suggestion
	opts := policyfragment.DefaultGetOperationOptions()
	opts.Format = &format
	resp, err := client.Get(ctx, *id, opts)
```